ursh/indico_ursh/plugin.py

```diff
@@ -0,0 +1,67 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+from __future__ import unicode_literals
+
+from flask_pluginengine import render_plugin_template
+from wtforms.fields.core import StringField
+from wtforms.fields.html5 import URLField
+from wtforms.validators import DataRequired
+
+from indico.core.plugins import IndicoPlugin, PluginCategory
+from indico.web.forms.base import IndicoForm
+from indico.web.views import WPBase
+
+from indico_ursh import _
+from indico_ursh.blueprint import blueprint
+
+
+class SettingsForm(IndicoForm):
+    api_key = StringField(_('API key'), [DataRequired()],
+                          description=_('The API key to access the <tt>ursh</tt> service'))
+    server = URLField(_('Shortening server'), [DataRequired()],
+                      description=_('The <tt>ursh</tt> shortening server (which generates short URLs)'))
+    host = URLField(_('Redirection host'), [DataRequired()],
+                    description=_('The <tt>ursh</tt> redirection host (which resolves short URLs to target)'))
+
+
+class UrshPlugin(IndicoPlugin):
+    """URL Shortener
+
+    Provides a URL shortening service for indico assets, events, etc.
+    """
+
+    configurable = True
+    settings_form = SettingsForm
+    category = PluginCategory.other
+
+    def init(self):
+        super(UrshPlugin, self).init()
+        self.template_hook('ursh', self._inject_ursh_button)
```

Comment:
this hook name is way too specific. let's find a name that suits the position where you put it (in that case you'd probably have quite a few differently named hooks) or use something like `url-shortener-link`. then it'd be clear what the hook is meant for, even when using it on completely different pages